### ISSUE TITLE
Difference between YOLO and JDE

### ISSUE THREAD

**Issue by IgnacioTsai** [WongKinYiu/yolor#29]
Does the difference of both layers do inference?
I saw the method

`io[..., :4] *= self.stride
            io[..., 4:] = F.softmax(io[..., 4:])`

What does it means?

**Comment by WongKinYiu** [comment]
Sorry about that I can not understand your question.

**Comment by IgnacioTsai** [comment]
> 
> 
> Sorry about that I can not understand your question.

No, I am the one should say sorry duo to weak expression.

My main question is "Does JDELayer is another prediction method of YOLO"?

The question what I means is the code from yolor/models/models.py , I saw that there is an hidden layer "JDELayer".
The keyword of "JDE" in YOLOR paper means "joint detection and embedding". I do not know what it means, but it looks like the structure of YOLOLayer, so that I compare with both code function in model.py.

I saw the only difference at the bottom of forward function.

YOLOLayer
`io = p.sigmoid()
io[..., :2] = (io[..., :2] * 2. - 0.5 + self.grid)
io[..., 2:4] = (io[..., 2:4] * 2) ** 2 * self.anchor_wh
io[..., :4] *= self.stride`

JDELayer
`io = p.clone()  # inference output
io[..., :2] = torch.sigmoid(io[..., :2]) * 2. - 0.5 + self.grid  # xy
io[..., 2:4] = (torch.sigmoid(io[..., 2:4]) * 2) ** 2 * self.anchor_wh  # wh yolo method
io[..., :4] *= self.stride
io[..., 4:] = F.softmax(io[..., 4:])`

It looks like different anchor predictions method, but I did not see any "jde" section in yolor relatived cfg.
That why I left this message.

By the way, thanks for reply.

**Comment by WongKinYiu** [comment]
yolo enable one box belongs to multiple classes, so it use sigmoid for decoding.
jde extract reid feature, an unique object must have only one id, so it use softmax for decoding.
it is better to take a look compute_loss function to understand the reason.
